### PR TITLE
refactor: change React.SFC to React.FC

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

With the release of hooks on React 16.8 (our package.json shows `"react": "^16.9.0"`), `React.SFC` has been marked as deprecated.

image

We should be using `FunctionComponent` instead.

This PR is basically a couple of changes:
- Rename `ReactSFCWithDisplayName` to `ReactFCWithDisplayName` (remove the 'S')
- Change all `React.SFC` to `React.FC`
- Rename `NamedSFC` to `NamedFC`

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
   - does not exist
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
   - not affected
- [x] Ran precheckin (`yarn precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above
   - not affected
- [x] (UI changes only) Verified usability with NVDA/JAWS
   - not affected

### CODE REVIEW

File: src/common/react/named-sfc.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@
 // Licensed under the MIT License.
 import * as React from 'react';
 
-export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };
+export type ReactFCWithDisplayName<P = {}> = React.FC<P> & { displayName: string };
 
-export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
+export function NamedSFC<P = {}>(displayName: string, component: React.FC<P>): ReactFCWithDisplayName<P> {
```
Comment (peterdur) [inline src/common/react/named-sfc.ts]: It's a broader change (130+ files) but do we want to consider renaming `NamedSFC` as well?

Comment (smoralesd) [reply src/common/react/named-sfc.ts]: I missed that (searched for react.sfc only)... I can do it as part of this PR or a separated one, any preference @peterdur ?

Comment (smoralesd) [reply src/common/react/named-sfc.ts]: done

File: src/DetailsView/components/adhoc-issues-test-view.tsx [hunk 2]
```diff
@@ -2,15 +2,15 @@
 // Licensed under the MIT License.
 import * as React from 'react';
 
-import { NamedSFC } from '../../common/react/named-sfc';
+import { NamedFC } from '../../common/react/named-sfc';
```
Comment (dbjorge) [inline src/DetailsView/components/adhoc-issues-test-view.tsx]: Should probably rename the file named-sfc.ts -> named-fc.ts to match

File: src/common/react/named-sfc.ts [hunk 3]
```diff
@@ -2,10 +2,10 @@
 // Licensed under the MIT License.
 import * as React from 'react';
 
-export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };
+export type ReactFCWithDisplayName<P = {}> = React.FC<P> & { displayName: string };
 
-export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
+export function NamedFC<P = {}>(displayName: string, component: React.FC<P>): ReactFCWithDisplayName<P> {
```
Comment (dbjorge) [inline src/common/react/named-sfc.ts]: I think there's a few instances of the string "SFC" in the tests for this that should probably also get updated

Comment (smoralesd) [reply src/common/react/named-sfc.ts]: updated

File: src/DetailsView/components/adhoc-issues-test-view.tsx (hunk 2 above)
Comment (smoralesd) [reply src/DetailsView/components/adhoc-issues-test-view.tsx]: updated.

**Pull request merged**